File: C++/PB3.cpp
/*
Teacher Took A Surprise Maths Test on Monday and only 5 students could score non-zero score and the maths 
teacher wants to find the runner-up among them
Give Input:
5
6 6 3 4 1
output: 
4
*/

#include<iostream>
using namespace std;

int check_similar_score(int score_list[], int n)
{

	if (n==0 || n==1)
	return n;
	
	int temp[n];
	
	int j = 0;
	int i;
	
	for (i=0; i<n-1; i++)
		if (score_list[i] != score_list[i+1])
			temp[j++] = score_list[i];
		temp[j++] = score_list[n-1];

	for (i=0; i<j; i++)
		score_list[i] = temp[i];
	
	return j;
}

int runnerup(int score_list[],int a){
	int j;
	for(j=1;j<a;j++){
		int key = score_list[j];
		int i = j-1;
		while(i>1 && score_list[i]>key){
			score_list[i+1] = score_list[i];
			i=i-1;
		}
		score_list[i+1] = key;
		return score_list[a-1];
	}
}

int main(){
	int n,i;
	int score_list[20];
	cin>>n;
	for(i=0;i<n;i++){
		cin>>score_list[i];
	}
	
	n = check_similar_score(score_list, n);
	
	int b = runnerup(score_list, n);
	
	cout<<b;
	
	return 0;
}

  

File: Python/PB3.py
'''
Teacher Took A Surprise Maths Test on Monday and only 5 students could score non-zero score and the maths 
teacher wants to find the runner-up among them
6 6 3 4 1
output: 
4
'''
def check_similar_score(score_list):
    for i in range(0,len(score_list)-1):
        n=score_list[i]
        if score_list.count(n)>1:
            score_list.remove(n)
    return score_list 
def runnerup(score_list,a):
    for j in range(1,a):
        key=score_list[j]
        i=j-1
        while (i>1) and (score_list[i]>key):
            score_list[i+1]=score_list[i]
            i=i-1
        score_list[i+1]=key
        return score_list[a-1]
        

if __name__ == '__main__':
    n = 5
    score_list=[6,6,3,4,1]
    score_list=check_similar_score(score_list)
    b=runnerup(score_list,len(score_list))
    print(b)